server/src/main/java/org/apache/uniffle/server/ShuffleTaskInfo.java

```diff
@@ -46,13 +47,20 @@ public class ShuffleTaskInfo {
 
   private AtomicReference<ShuffleDataDistributionType> dataDistType;
 
+  private AtomicLong totalDataSize = new AtomicLong(0);
+  /**
+   * shuffleId -> partitionId -> partition shuffle data size
+   */
+  private Map<Integer, Map<Integer, Long>> partitionDataSizes;
```

Comment:
AtomicLong?

Reply:
It will be added by the `ConcurrentHashMap.computeIfPresent,` this is a thread safe operation. Right?

Reply:
What will it happen if two threads add the same partition?

Reply:
This will guarded by `computeIfPresent` to keep thread safe when having multiple threads.

Reply:
You can see the code of `computeIfpresent`.  `computeIfpresent` don't  guard the atomicity. Only the one partition will be added.

Reply:
https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/ConcurrentHashMap.html#computeIfPresent-K-java.util.function.BiFunction-

Reply:
OK, my mistake. I just see the default method of `Map`.